src/Collection.php

```diff
@@ -28,13 +29,15 @@ public function __construct($input)
     {
         if (is_callable($input)) {
             $this->inputFactory = $input;
-            $this->input = $input();
-        } elseif (is_array($input)) {
+            $input = $input();
+        }
+
+        if (is_array($input)) {
             $this->input = new RecursiveArrayIterator($input);
```

Comment:
Do we even need a `RecursiveArrayIterator` here? Or would an `ArrayIterator` suffice? Is there any difference apart from the additional `{get|has}Children()` methods in the recursive one (which we don't use)?